keyboards/ashpil/modelm_usbc/modelm_usbc.c

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2019 ashpil
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www./gnu.org/licenses/>.
+ */
+#include "modelm_usbc.h"
+
+void keyboard_pre_init_kb(void) {
+  /* Setting status LEDs pins to output and +5V (off) */
+  setPinOutput(D5);
+  setPinOutput(D6);
+  setPinOutput(D7);
+  writePinHigh(D5);
+  writePinHigh(D6);
+  writePinHigh(D7);
+}
+
+void matrix_init_kb(void) {
+  /* put your keyboard start-up code here
+   * runs once when the firmware starts up */
+
+  matrix_init_user();
+}
+
+void matrix_scan_kb(void) {
+  /* put your looping keyboard code here
+   * runs every cycle (a lot) */
+
+  matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  /* put your per-action keyboard code here
+   * runs for every action, just before processing by the firmware */
+
+  return process_record_user(keycode, record);
+}
+
+void led_set_kb(uint8_t usb_led) {
+  if (usb_led & (1<<USB_LED_NUM_LOCK)) {
+    writePinLow(D5);
+  } else {
+    writePinHigh(D5);
+  }
+  if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+    writePinLow(D6);
+  } else {
+    writePinHigh(D6);
+  }
+  if (usb_led & (1<<USB_LED_SCROLL_LOCK)) {
+    writePinLow(D7);
+  } else {
+    writePinHigh(D7);
+  }
+
+  led_set_user(usb_led);
+}
```

Comment:
Could you change this function to the following:

```c
bool led_update_kb(led_t led_state) {
    if(led_update_user(led_state)) {
        writePin(D5, !led_state.num_lock);
        writePin(D6, !led_state.caps_lock);
        writePin(D7, !led_state.scroll_lock);
    }
    return true;
}
```

This will allow user keymaps to override it with their own code, for example for layer indication.